src/bp/core/routers/admin/languages.ts

```diff
@@ -144,23 +146,27 @@ export class LanguagesRouter extends CustomRouter {
       '/available',
       this.needPermissions('read', this.resource),
       this.asyncMiddleware(async (req, res) => {
+        let languagesData
         try {
           const client = await this.getSourceClient()
-          const { data } = await client.get('/languages')
+          languagesData = (await client.get('/languages')).data
+        } catch (e) {
+          languagesData = { installed: [], available: [] }
+          const { languageSources } = await this.ghostService.global().readFileAsObject('/config/', 'nlu.json')
+          if (languageSources.length && languageSources[0].endpoint)
+            this.logger.warn(`Please remove the languageSources from nlu.json if you don't want to use it`)
```

Comment:
You don't need the ghost service to get the language sources, check the method getSourceClient, which uses the module loader

Reply:
`this.moduleLoader.configReader.getGlobal('nlu')` throws an error in case NLU is disabled. Or am I missing smth?

Reply:
`this.moduleLoader.configReader.getGlobal('nlu')` throws an error in case NLU is disabled. Or am I missing smth?

Reply:
That's even better. Using the ghost you don't know if it's enabled or not

Reply:
Oh OK, I thought you meant we need to display it even if NLU is disabled.

Reply:
Done! Let me know if you see any other issues.